Gem/Code/Source/Components/UI/UiStartMenuComponent.cpp

```diff
@@ -0,0 +1,127 @@
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &UiStartMenuComponent::m_connectToHostFailedUi, "Connection To Host Failed", "Ui to inform the user that connecting to the host failed.")
+                    ;
+            }
+        }
+    }
+
+    void UiStartMenuComponent::Activate()
+    {
+        UiCursorBus::Broadcast(&UiCursorInterface::IncrementVisibleCounter);
+
+        // Listen for button presses
+        UiButtonBus::Event(m_exitButtonUi, &UiButtonInterface::SetOnClickCallback, [this](AZ::EntityId buttonEntityId, [[maybe_unused]] AZ::Vector2 position) {OnButtonClicked(buttonEntityId); });
+        UiButtonBus::Event(m_hostButtonUi, &UiButtonInterface::SetOnClickCallback, [this](AZ::EntityId buttonEntityId, [[maybe_unused]] AZ::Vector2 position) {OnButtonClicked(buttonEntityId); });
+        UiButtonBus::Event(m_joinButtonUi, &UiButtonInterface::SetOnClickCallback, [this](AZ::EntityId buttonEntityId, [[maybe_unused]] AZ::Vector2 position) {OnButtonClicked(buttonEntityId); });
+        UiButtonBus::Event(m_connectToHostFailedUi, &UiButtonInterface::SetOnClickCallback, [this](AZ::EntityId buttonEntityId, [[maybe_unused]] AZ::Vector2 position) {OnButtonClicked(buttonEntityId); });
+
+        // Hide the host button if this app can't host
+        #if !AZ_TRAIT_SERVER
+            UiElementBus::Event(m_hostButtonUi, &UiElementInterface::SetIsEnabled, false);
+        #endif
+
+        // Hide the attempting connection ui until the player tries to connect
+        UiElementBus::Event(m_attemptConnectionBlockerUi, &UiElementInterface::SetIsEnabled, false);
+
+        // Set default text of the ip-address
+        UiTextInputBus::Event(m_ipAddressTextInputUi, &UiTextInputInterface::SetText, Multiplayer::LocalHost);
+
+        // Listen for disconnect events to know if connecting to the host server failed
+        AZ::Interface<Multiplayer::IMultiplayer>::Get()->AddEndpointDisconnectedHandler(m_onConnectToHostFailed);
+    }
+
+    void UiStartMenuComponent::Deactivate()
+    {
+        m_onConnectToHostFailed.Disconnect();
+        UiCursorBus::Broadcast(&UiCursorInterface::DecrementVisibleCounter);
+    }
+
+    void UiStartMenuComponent::OnButtonClicked(AZ::EntityId buttonEntityId) const
+    {
+        const auto console = AZ::Interface<AZ::IConsole>::Get();
+        if (!console)
+        {
+            AZ_Assert(false, "UiStartMenuComponent attempting to use console commands before AZ::Console is available.");
+            return;
+        }
+
+        if (buttonEntityId == m_exitButtonUi)
+        {
+            console->PerformCommand("quit");
+        }
+        else if (buttonEntityId == m_hostButtonUi)
+        {
+            console->PerformCommand("host");
+            console->PerformCommand("loadlevel NewStarbase");
```

Comment:
Really feels like the next level to load should at least be a parameter on the component.

Reply:
Originally had that as a variable in the UI, but felt like a variable I wouldn't want art to accidentally change/misspell. Maybe a class static constexpr would be more appropriate.

Reply:
fixed!

Reply:
"Originally had that as a variable in the UI, but felt like a variable I wouldn't want art to accidentally change/misspell." - you can make it an Asset Reference that can only be set to a Spawnable type. This solves the misspelling problem and also gives the asset dependency system more information for creating bundles.